src/VerifyResourceUsage.pl

```diff
@@ -341,45 +245,87 @@ sub FindDirs {
         }
         close(FIL) or die "can't close $file: $!";
 
-        # find all call sites and EnsureArgCountCorrect
+        # find all call sites
         my $rest = $filestring;
 
         while ($rest =~ /LogMessages\.(\w+)/g)
         {   
             my $id = $1;
 
-            if (defined $ids{$id})   # sometimes there will be commented-out code with LogMessages.ThisIDDoesntExistAnymore
+            if (defined $allids{$id})
+            {
+                $allids{$id}++;   # increment the number of references to this id
+            }
+            else
             {
-                $ids{$id}[0]++;   # increment number-references-to-this-id
+                $allids{$id} = 1;
             }
         }
     }
-
     PrintPerf("\nPERF: done reading and processing source files\n");
-    {
-        PrintPerf("\nDone processing files\n");
+}
+
+# go through all the directories again to discover unused, mismatching, and/or duplicate log message ids
+foreach my $directory (keys(%dirhash)) 
```

Comment:
would it make sense to also print the dirhash?

Reply:
This script prints out the directory a log message file belongs to when it processes it, so all the keys in the dirhash are eventually printed. Printing out the entire dirhash would result in all the corresponding log messages being printed out as well, which I think just ends up being too much information.